### PR TITLE
refactor(Signaling): Added send function

### PR DESCRIPTION
**Pull request by pchalupa**

### CODE REVIEW

File: src/context/Chat/index.tsx [hunk 1]
```diff
@@ -247,6 +226,21 @@ export const ChatProvider = ({ children, signalingServer, iceServers }: Props) =
...
+	};
+
+	const handleSignalingOpen = () => {
+		sendSignalingMessage('all', { displayName: 'aaa' });
+	};
+
+	useEffect(() => {
+		signaling.current.addEventListener('message', gotMessageFromServer);
+		signaling.current.addEventListener('open', handleSignalingOpen);
+	}, [signaling]);
```
Comment (pchalupa) [inline src/context/Chat/index.tsx:242]: I am not sure with this.

File: src/context/Chat/index.tsx [hunk 2]
```diff
@@ -47,12 +47,12 @@ export const ChatContext = createContext<ContextType>(contextDefaults);
...
 export const ChatProvider = ({ children, signalingServer, iceServers }: Props) => {
 	const [isEntered, setIsEntered] = useState(false); // TODO: Rename, leave there
-	const [localUuid] = useState(uuid()); // TODO: Check ref, state, or global variable
+	const localUuid = useRef(uuid()).current; // TODO: Remove current
 	const [messageData, setMessageData] = useState<MessageData[]>([]); // TODO: Leave, check interface, remove: avatar, event, username
 	const [error, setError] = useState<string>(''); //TODO: Remove error, use onError event instead
 	const [user, setUser] = useState({ name: 'test', avatar: '' }); // TODO: Remove
-	// TODO: Rename to signaling
-	const signaling = useRef<WebSocket>(null);
+	const signaling = useRef<WebSocket>(new WebSocket(signalingServer));
```
Comment (pchalupa) [inline src/context/Chat/index.tsx:54]: We are going to connect to socket right away. Is it right?

File: src/index.tsx [hunk 3]
```diff
@@ -6,13 +6,10 @@ import App from './App';
...
 
 ReactDOM.render(
-  <ChatProvider
-    signalingServer='ws://3.71.110.139:8001/'
-    iceServers={[{ urls: 'stun:stun.l.google.com:19302' }]}
-  >
-    <App />
-  </ChatProvider>,
-  document.getElementById('root'),
+	<ChatProvider signalingServer="ws://localhost:8001/" iceServers={[{ urls: 'stun:stun.l.google.com:19302' }]}>
```
Comment (pchalupa) [inline src/index.tsx:9]: We are going to use local server.

File: src/context/Chat/index.tsx (hunk 2 above)
Comment (BrantalikP) [reply src/context/Chat/index.tsx:54]: Hmm, I was looking for something about this approach and I am not really sure.. But by my opinion would we be better to have init state `null` Maybe there will be some cases when we won't do not want to have signaling active right away. (Waiting for authentication or something like that)

File: src/context/Chat/index.tsx (hunk 1 above)
Comment (BrantalikP) [reply src/context/Chat/index.tsx:242]: should not we also remove the listeners on onmounting?

Comment (pchalupa) [reply src/context/Chat/index.tsx]: Done

File: src/context/Chat/index.tsx (hunk 2 above)
Comment (pchalupa) [reply src/context/Chat/index.tsx]: Change reverted

**Pull request merged**